Source/Core/S2Cell.js

```diff
@@ -0,0 +1,589 @@
+};
+
+/**
+ * Gets the parent cell at the given level.
+ *
+ * @returns {S2Cell} Returns the parent of the S2Cell.
+ * @private
+ */
+S2Cell.prototype.getParentAtLevel = function (level) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._level === 0 || level < 0 || this._level < level) {
+    throw new DeveloperError("cannot get parent at invalid level.");
+  }
+  //>>includeEnd('debug');
+  var newLsb = lsbForLevel(level);
+  return new S2Cell((this._cellId & -newLsb) | newLsb);
+};
+
+/**
+ * @private
+ */
+S2Cell.fromFacePosLevel = function (face, pos, level) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.bigint("pos", pos);
+  if (face < 0 || face > 5) {
+    throw new DeveloperError("Invalid S2 Face (must be within 0-5)");
+  }
+
+  if (level < 0 || level > S2MaxLevel) {
+    throw new DeveloperError("Invalid S2 Face (must be within 0-5)");
+  }
+  if (pos < 0 || pos > Math.pow(4, level)) {
+    throw new DeveloperError("Invalid Hilbert position for level");
+  }
+  //>>includeEnd('debug');
+  var cell = new S2Cell((face << BigInt(S2PositionBits)) + (pos | BigInt(1))); // eslint-disable-line
+  return cell.getParentAtLevel(level);
+};
+
+/**
+ * Get center of the S2 cell.
+ *
+ * @returns {Cartesian} The position of center of the S2 cell.
+ * @private
+ */
+S2Cell.prototype.getCenter = function () {
+  var center = getS2Center(this._cellId);
+  // Normalize XYZ.
+  Cartesian3.normalize(center, center);
+  var cartographic = new Cartographic.fromCartesian(
+    center,
+    Ellipsoid.UNIT_SPHERE
+  );
+  // Interpret spherical coordinates on UNIT_SPHERE as cartographics on WGS84.
+  return Cartographic.toCartesian(
+    cartographic,
+    Ellipsoid.WGS84,
+    new Cartesian3()
+  );
+};
+
+/**
+ * Get vertex of the S2 cell.
+ *
+ * @param {Number} index An integer index of the vertex.
+ * @returns {Cartesian} The position of the vertex of the S2 cell.
+ * @private
+ */
+S2Cell.prototype.getVertex = function (index) {
+  var vertex = getS2Vertex(this._cellId, index);
+  // Normalize XYZ.
+  Cartesian3.normalize(vertex, vertex);
+  var cartographic = new Cartographic.fromCartesian(
+    vertex,
+    Ellipsoid.UNIT_SPHERE
+  );
+  // Interpret spherical coordinates on UNIT_SPHERE as cartographics on WGS84.
+  return Cartographic.toCartesian(
+    cartographic,
+    Ellipsoid.WGS84,
+    new Cartesian3()
+  );
+};
+
+/**
+ * @private
+ */
+function getS2Center(cellId) {
+  var faceSiTi = CellIdToFaceSiTi(cellId);
+  return FaceSiTitoXYZ(faceSiTi[0], faceSiTi[1], faceSiTi[2]);
+}
+/**
+ * @private
+ */
+function getS2Vertex(cellId, index) {
+  var faceIJ = CellIdToFaceIJ(cellId);
+  var uv = IJLeveltoBoundUV([faceIJ[1], faceIJ[2]], S2Cell.getLevel(cellId));
+  var y = (index >> 1) & 1;
+  return FaceUVtoXYZ(faceIJ[0], uv[0][y ^ (index & 1)], uv[1][y]);
+}
+
+// S2 Coordinate Conversions
+
+/**
+ * @private
+ */
+function CellIdToFaceSiTi(cellId) {
```

Comment:
Naming conventions, functions should always be `camelCase`  https://github.com/CesiumGS/cesium/blob/master/Documentation/Contributors/CodingGuide/README.md#naming